docs/developer_docs/front_end_code.rst

```diff
@@ -9,6 +9,51 @@ Inline Javascript (i.e. Javascript directly in the Django templates inside `<scr
 
 For templating on the front end, we use `Handlebars.js <http://handlebarsjs.com/>`_ to render templates with a restricted set of statements and access to all variables passed into the template context.
 
+Modularity
+----------
+
+In order to maintain modular code and be explicit about our dependencies, we use Browserify to build Javascript code into bundles for use on the client side.
+
+To specify a bundle to be imported into the page, you need to create a 'bundle module' - this will be automatically detected by our Javascript build script, and be built into a bundle that can then be included as a script tag in a Django template.
+
+'Bundle modules' are specified inside the static/js directory of a Django app - e.g. 'bundle modules' in distributed are under ``kalite/distributed/static/js/distributed/bundle_modules``. Here is a simple example of a bundle_module::
+
+    var $ = require("base/jQuery");
+    var TopicChannelRouter = require("topics/router");
+    var Backbone = require("base/backbone");
+
+    module.exports = {
```

Comment:
`module.exports` is a new idiom for me. Can you describe this in detail, or perhaps link to a `browserify` tutorial where we can read more about it?